### PR TITLE
Fix wasm codegen for i32_to_double_s and i32_to_double_u

### PR DESCRIPTION
**Pull request by steven-johnson**
These were also easily fixed by using LLVM builtins.

### CODE REVIEW

File: src/runtime/wasm_math.ll [hunk 1]
```diff
@@ -140,28 +140,15 @@ define weak_odr <8 x i16> @saturating_narrow_i32x8_to_u16x8(<8 x i32> %x) nounwi
 }
 
 ; Integer to double-precision floating point
+define weak_odr <4 x double> @i32_to_double_s(<4 x i32> %x) nounwind alwaysinline {
+  %1 = sitofp <4 x i32> %x to <4 x double>
```
Comment (dsharletg) [inline src/runtime/wasm_math.ll:144]: This is almost certainly what CodeGen_LLVM does. Are you sure we need these helpers?

Comment (steven-johnson) [reply src/runtime/wasm_math.ll:144]: Without these helpers, everything gets scalarized.

Comment (steven-johnson) [reply src/runtime/wasm_math.ll:144]: (But yeah, this is definitely suboptimal. I'll follow up with wasm LLVM folks to see if this can be improved.)

Comment (dsharletg) [reply src/runtime/wasm_math.ll:144]: Why does it scalarize? That doesn't make any sense.

Comment (steven-johnson) [reply src/runtime/wasm_math.ll:144]: You are correct. It doesn't make any sense. I'm contacting the relevant wasm person right now.

**Review by dsharletg — APPROVED** [review]

File: src/runtime/wasm_math.ll (hunk 1 above)
Comment (steven-johnson) [reply src/runtime/wasm_math.ll:144]: Looking more closely at the code, these cases actually *don't* get scalarized, but neither to they generate the opcodes we expect; instead, it generates a sequence of ops that loads two int32s, extends them to int64, then does bit-twiddling to convert them to IEEE754 doubles. Maybe there's a codegen path that predates these instructions being added to the wasmSimd128 spec that hasn't been updated? I'll circle around with the wasm-llvm folks to see what's going on. In the meantime, this is arguably an improvement on the status quo.

Comment (steven-johnson) [reply src/runtime/wasm_math.ll:144]: I'm going to leave this open (not landed) pending the LLVM investigation.

Comment (dsharletg) [reply src/runtime/wasm_math.ll:144]: Did you also report this for float -> double, from #6132?

Comment (steven-johnson) [reply src/runtime/wasm_math.ll:144]: I'm going to provide repro cases for that as well

Comment (steven-johnson) [reply src/runtime/wasm_math.ll:144]: Opened https://bugs.llvm.org/show_bug.cgi?id=51098

**Comment by steven-johnson** [comment]
This was fixed in upstream LLVM, so I'm closing this.

**Pull request closed without merging**